wiring/inc/spark_wiring_ble.h

```diff
@@ -0,0 +1,460 @@
+    BleAddress address;
+    BleConnectionParams connParams;
+    BleConnectionHandle connHandle;
+    int8_t rssi;
+
+    BlePeerDevice();
+    BlePeerDevice(const BleAddress& addr);
+    ~BlePeerDevice();
+
+    BleCharacteristic characteristic(const char* desc) const;
+    BleCharacteristic characteristic(const BleUuid& uuid) const;
+
+    bool connected(void) const;
+
+    BleGattServerImpl* gattsProxy(void) {
+        return gattsProxy_.get();
+    }
+
+    bool operator == (const BlePeerDevice& device);
+
+private:
+    std::shared_ptr<BleGattServerImpl> gattsProxy_;
+};
+
+
+class BleLocalDevice {
+public:
+    BleAddress address;
+
+    int on(void);
+    void off(void);
+
+    int setTxPower(int8_t val) const;
+    int8_t txPower(void) const;
+
+    int advertise(void);
+    int advertise(BleAdvertisingData* advertisingData, BleAdvertisingData* scanResponse=nullptr);
+    int advertise(uint16_t interval);
+    int advertise(uint16_t interval, BleAdvertisingData* advertisingData, BleAdvertisingData* scanResponse=nullptr);
+    int advertise(uint16_t interval, uint16_t timeout);
+    int advertise(uint16_t interval, uint16_t timeout, BleAdvertisingData* advertisingData, BleAdvertisingData* scanResponse=nullptr);
+    int advertise(const BleAdvertisingParams& params);
+    int advertise(const BleAdvertisingParams& params, BleAdvertisingData* advertisingData, BleAdvertisingData* scanResponse=nullptr);
+
+    int advertise(const iBeacon& iBeacon, bool connectable=false);
+    int advertise(uint16_t interval, const iBeacon& iBeacon, bool connectable=false);
+    int advertise(uint16_t interval, uint16_t timeout, const iBeacon& iBeacon, bool connectable=false);
+    int advertise(const BleAdvertisingParams& params, const iBeacon& iBeacon, bool connectable=false);
+
+    int stopAdvertising(void) const;
+
+    int scan(OnScanResultCallback callback);
+    int scan(OnScanResultCallback callback, uint16_t timeout);
+    int scan(BleScanResult* results, size_t resultCount);
+    int scan(BleScanResult* results, size_t resultCount, uint16_t timeout);
+    int scan(BleScanResult* results, size_t resultCount, const BleScanParams& params);
+    int stopScanning(void);
+
+    int addCharacteristic(BleCharacteristic& characteristic);
+    int addCharacteristic(const char* desc, BleCharacteristicProperties properties, OnDataReceivedCallback callback=nullptr);
+
+    template<typename T>
+    int addCharacteristic(const char* desc, BleCharacteristicProperties properties, T charUuid, T svcUuid, OnDataReceivedCallback callback=nullptr) {
+        BleCharacteristic characteristic(desc, properties, charUuid, svcUuid, callback);
+        return addCharacteristic(characteristic);
+    }
+
+    int setPPCP(uint16_t minInterval, uint16_t maxInterval, uint16_t latency=BLE_DEFAULT_SLAVE_LATENCY, uint16_t timeout=BLE_DEFAULT_CONN_SUP_TIMEOUT);
+
+    BlePeerDevice connect(const BleAddress& addr, uint16_t interval, uint16_t latency, uint16_t timeout);
```

Comment:
`system_tick_t` ?

Reply:
I'd suggest not. The Bluetooth specification says that these parameters are 2-octets value, while the `system_tick_t` is `uint32_t` actually.